daemon/cmd/daemon.go

```diff
@@ -91,6 +91,8 @@ const (
 // Daemon is the cilium daemon that is in charge of perform all necessary plumbing,
 // monitoring when a LXC starts.
 type Daemon struct {
+	lock.RWMutex // This mutex used to synchronize test setups. It is not used for implementation code.
```

Comment:
I would rather have this lock be explicitly called as `testLock` or something similar, this will make sure that we don't lose track of getting rid of it at some point and will make anyone who tries to use it outside of tests think twice (it will also be easier to spot these occurences in reviews).

Reply:
I've found a way to avoid using a mutex: bb4712c